roles/neutron-common/templates/etc/neutron/neutron_lbaas.conf

```diff
@@ -3,9 +3,16 @@
 [quotas]
 
 [service_providers]
+{% if neutron.lbaas.f5.enabled | bool %}
+service_provider = {{ neutron.lbaas.f5.service_provider }}
+{% else %}
 service_provider = {{ neutron.lbaas.service_provider }}
+{% endif %}
```

Comment:
move lbaas.service_provider (one that is for haproxy above) under lbaas.haproxy.service_provider .. Will need to make appropriate changes in tasks/conf files.